tests/test.cpp

```diff
@@ -598,16 +598,18 @@ TEST_CASE("K Sizes")
 
     SECTION("Minimum K Size")
     {
-        uint8_t proof_data[0];
+        uint8_t *proof_data = new uint8_t[0];
```

Comment:
you're not allowed to allocate 0-length arrays. Both this and the previous version are invalid.
Don't you know the size of this buffer?
You could probably just do something like:
```
uint8_t proof_data[32];
```
and pass a pointer to that.